WooCommerce/Classes/POS/Models/PointOfSaleAggregateModel.swift

```diff
@@ -215,10 +218,9 @@ extension PointOfSaleAggregateModel {
 
     @MainActor
     func cancelCashPayment() async {
+        paymentState = .card(.idle)
```

Comment:
Interesting one, as only happens with free orders. I didn't thought about this one when implementing the cancellation since we didn't support those, but a good case to add to future testing charts nonetheless as we were planning to 👍

Reply:
Yeah, it was surprising to me, but it's one of those cases where our state modelling doesn't _quite_ match reality, because we don't keep hold of the card reader state.

That means we need to juggle it ourself, in this case remembering to set the card reader state back to something else.

We could improve this by always keeping both card and cash payment statuses (adding `cash(.idle)`), and then having something which indicates the active payment type.

It would be more obvious when cancelling the cash payment that we needed to change the `activePaymentMethod`